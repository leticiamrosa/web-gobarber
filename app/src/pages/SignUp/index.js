import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { Form, Input } from '@rocketseat/unform';
import * as Yup from 'yup';

import { signUpRequest } from '~/store/modules/auth/actions';

import logo from '~/assets/logo.svg';

const schema = Yup.object().shape({
  name: Yup.string().required('O nome é obrigatório'),
  email: Yup.string().email('Insira um e-mail válido').required('O email é obrigatório'),
  password: Yup.string().required('A senha é obrigatória'),
})

export default function SignUp({ t }) {
  const dispatch = useDispatch();

  function handleSubmit({ name, email, password} ) {
    dispatch(signUpRequest(name, email, password));
  }

  return (
    <>
      <img src={logo} alt="GoBarber" />

      <Form schema={schema} onSubmit={handleSubmit}>
        <Input name="name" placeholder={t('Nome completo')} />
        <Input name="email"type="email" placeholder={t('Seu e-mail')} />
        <Input name="password" type="password" placeholder={t('Sua senha secreta')} />
        <button type="submit">{t('Criar conta')}</button>
        <Link to="/">{t('Já tenho login')}</Link>
      </Form>
    </>
  );
}
SignUp.propTypes = {
  t: PropTypes.func.isRequired,
};

