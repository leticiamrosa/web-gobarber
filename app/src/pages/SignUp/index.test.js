import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SignUp from './index';

const setup = propOverrides => {
  const props = Object.assign({
    t: jest.fn()
  },propOverrides);

  const result = shallow(<SignUp {...props} />);

  return {
    props,
    result,
  };
};

describe('SignUp component', () => {

  it('should renders component without crashing', () => {
    //given
    //when
    const { result } = setup();
    //then
    expect(toJson(result)).toMatchSnapshot();
  })

  describe('component renders with props', () => {

    it('when received i18n props', () => {
    //given
    const props = {

    }
    //when
    const { result } = setup();
    const expectedElement = result.find('form').find('#email');
    console.log(expectedElement.debug())

    //then
    // expect(expectedElement.exists()).toBe(true);
  })
  });

});


describe('renders inputs', () => {

  it('should renders email input', () => {
    //given
    const props = {
      email: 'leticia@dev.com'
    }
    //when
    const { result } = setup(props);
    console.log(result.find('email'))
    //then
    // expect(result.find('email'))
  })
});

