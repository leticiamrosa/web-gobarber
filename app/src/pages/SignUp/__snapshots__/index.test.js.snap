// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should renders component without crashing 1`] = `
<Fragment>
  <img
    alt=""
    src="logo.svg"
  />
  <form>
    <input />
    <input
      type="email"
    />
    <input
      type="password"
    />
    <button
      type="submit"
    />
    <Link
      to="/"
    />
  </form>
</Fragment>
`;
